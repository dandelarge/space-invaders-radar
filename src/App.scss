@import './styles/animations';

.App {
    margin: 30px;
    display: flex;
    justify-content: space-around;

    h3 {
        border: 1px solid #ddd;
        background-color: white;
        color: black;
        max-width: 200px;
        padding: 30px 15px;
        font-size: larger;
    }

    .radarScreen  {
        background-color: black;
        background-image: radial-gradient(
          rgba(0, 150, 0, 0.4), black 120%
        );
        margin: 0;
        overflow: hidden;
        padding: 2rem;
        &#radarInput {
            width: 620px;
            height: 620px;
        }
        .textarea {
            border: none;
            color: greenyellow;
            background-color: transparent;
            font: 12px Inconsolata, monospace;
            text-shadow: 0 0 5px #C8C8C8;
            width: 100%;
            height: 100%;
            &.radarInput {
                animation: glowingText 3s cubic-bezier(1, 0.72, 0.165, 0.1) 1s infinite;
            }

            &.larger-text {
                font-size: larger;
            }
        }

        &::after {
          content: "";
          position: absolute;
          top: 0;
          left: 0;
          width: 100vw;
          height: 100vh;
          background: repeating-linear-gradient(
            0deg,
            rgba(black, 0.15),
            rgba(black, 0.15) 1px,
            transparent 1px,
            transparent 2px
          );
          pointer-events: none;
        }
      }


      ::selection {
        background: #0080FF;
        text-shadow: none;
      }


      pre {
        margin: 0;
      }



}