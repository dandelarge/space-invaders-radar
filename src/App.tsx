import React, { useEffect, useReducer } from 'react';
import './App.scss';
import { globals } from './config';
import { Spider } from './core/invaders/Spider';
import { Squid } from './core/invaders/Squid';
import { RadarState } from './core/radar.types';
import { Radar } from './core/Radar';
import { radarReducer } from './core/reducers/radar-reducer';
import Results from './components/results/results';

const spider = new Spider();
const squid = new Squid();

export const RadarDispatch = React.createContext({});

const initialState: RadarState = {
  input: globals.DEFAULT_READING,
  invaders: [spider,squid],
  samples: [],
  results: [],
  definition: 0.8
}

function radarInit(state: RadarState) {
  Radar.processInput(state);
  return {...state};
}

function App() {
  const [state, dispatch] = useReducer(radarReducer, initialState, radarInit);
  useEffect(() => {
    dispatch({type: 'PROCESS_INPUT', newState: state});
  }, [])

  function onProcessButtonClick() {
    dispatch({type: 'PROCESS_INPUT', newState: state });
  }
  function onInputChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
    const input = event.target.value;
    dispatch({type: 'UPDATE_INPUT', newState:{...state, input}});
  }

  function onCleanupButtonClick () {
    const cleanInput = Radar.cleanNoise(Radar.processImage(state.input), state.results);
    dispatch({type: 'UPDATE_STATE', newState: {...state, input: cleanInput}});
  }

  return (
    <div className="App">
      <RadarDispatch.Provider value={dispatch}>
        <section className="input-panel">
          <span>match ratio</span>
          <input type="number" max="1" placeholder="0.8" value={state.definition}
            onChange={event => dispatch({type: 'UPDATE_STATE', newState: {...state, definition: parseFloat(event.target.value)}})}/>
          <div id="radarInput" className="radarScreen" spellCheck="false">
            <textarea className="textarea radarInput" value={state.input} onChange={onInputChange}></textarea>
          </div>
          <button onClick={onProcessButtonClick}>process Input</button>
          <button onClick={onCleanupButtonClick}>Clean noise</button>
        </section>
        <div className="results-panel">
          <Results state={state}></Results>
        </div>
      </RadarDispatch.Provider>
    </div>
  );
}

export default App;
