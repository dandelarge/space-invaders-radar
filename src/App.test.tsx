import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

test('renders App component', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
