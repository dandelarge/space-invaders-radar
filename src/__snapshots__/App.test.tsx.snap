// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component 1`] = `
<div
  className="App"
>
  <ContextProvider
    value={[Function]}
  >
    <section
      className="input-panel"
    >
      <span>
        match ratio
      </span>
      <input
        max="1"
        onChange={[Function]}
        placeholder="0.8"
        type="number"
        value={0.8}
      />
      <div
        className="radarScreen"
        id="radarInput"
        spellCheck="false"
      >
        <textarea
          className="textarea radarInput"
          onChange={[Function]}
          value="
----o--oo----o--ooo--ooo---------o---oo-o----oo---o--o---------o----o------o----------------o--o--o-
--o-o-----oooooooo-oooooo-------o----o------ooo-o---o--o----o------o--o---ooo-----o--oo-o------o----
--o--------oo-ooo-oo-oo-oo------------------ooooo-----o-----o------o---o--o--o-o-o------o----o-o-o--
-------o--oooooo--o-oo-o--o-o-----oo--o-o-oo--o-oo-oo-o--------o-----o------o-ooooo---o--o--o-------
------o---o-ooo-ooo----o------o-------o---oo-ooooo-o------o----o--------o-oo--ooo-oo-------------o-o
-o--o-----o-o---o-ooooo-o-------o----o---------o-----o-oo-----------oo----ooooooo-ooo-oo------------
o-------------ooooo-o--o--o--o-------o--o-oo-oo-o-o-o----o-------------o--oooo--ooo-o----o-----o--o-
--o-------------------------oo---------oo-o-o--ooo----o-----o--o--o----o--o-o-----o-o------o-o------
-------------------o-----------------o--o---------------o--------o--oo-o-----oo-oo---o--o---o-----oo
----------o----------o------------------o--o----o--o-o------------oo------o--o-o---o-----o----------
------o----o-o---o-----o-o---------oo-o--------o---------------------------------o-o-o--o-----------
---------------o-------------o-------o-------------------o-----o---------o-o-------------o-------oo-
-o--o-------------o-o--------o--o--oo-------------o----ooo----o-------------o----------oo----o---o-o
-o--o-------------o----oo------o--o-------o--o-----------o----o-----o--o----o--oo-----------o-------
-o-----oo-------o------o---------------o--o----------o-----o-------o-----------o---o-o--oooooo-----o
-o--------o-----o-----o---------oo----oo---o-----------o---o--oooo-oo--o-------o------oo--oo--o-----
------------o-------------------o----oooo-------------oo-oo-----ooo-oo-----o-------o-oo-oooooooo---o
-----------------------------------oooooooo---o-----o-------o--oooooo-o------------o-o-ooooooo-o----
------------o------o-------o-------oo-oo--o--o---------o--o-o-o-ooooo-o--------------oo-o----o-oo-o-
---o-o----------o--------oo----o----oooooooo-------o----o-o-o-o-----o-o-----o----------ooo-oo--o---o
-o-o---------o-o---------------o--o--o--ooo---ooo-------o------oo-oo------------o--------o--o-o--o--
-------oo---------------------------o-oo----------o------o-o-------o-----o----o-----o-oo-o-----o---o
---o--------o-----o-------o-oo-----oo--oo-o----oo----------o--o---oo------oo----o-----o-------o-----
---o--ooo-o---------o-o----o------------o---------o----o--o-------o-------------o----------------oo-
---o------o----------------o----o------o------o---oo-----------o-------------o----------oo---------o
--oo---------------o--o------o---o-----o--o-------------o------o-------o-----o-----o----o------o--o-
-o-------o----------o-o-o-------o-----o--o-o-----------o-oo-----------o------o---------o-----o-o----
----------o----o-------o----o--o------o------------o---o---------------oo----o-----ooo--------------
----o--------oo----o-o----o--o------ooo----o-oooo---o--o-oo--------o-oo-----o-o---o-o--o-----oo-----
------o--------o-ooooo----o---o--o-----o---------------o-o-------o-----o----------------------------
o-------oo----o--oooooo-o---o--o------oooo----------o-oo-------o---o----------o------oo-------------
-o---o----------o--oo-oo-o---o-----o-o-----------------------oo--o------o------o--------------------
-----oo-o-o-o---ooooooooo----o----o--------o--o---oo---o------------o----------o-o---o------o-o--oo-
------o------o---ooo-o---------------------------o--o---o---o----o--o-------o-----o------o----o----o
-------o----------ooo-o-----o----o---o--o-oo--o--o-o--o------o--o-oo---ooo------------------------o-
-o-------o------o-o--ooo--o---o---oo-----o----o-------------o----o-ooo-o------o--o-o------o-o-------
---oo--o---o-o---------o---o--------------o--o-----o-------o-----o--o---o-oo--------o----o----o-----
o------o----oo-o-----------oo--o---o--------o-o------o-------o-o------o-oo---------o-----oo---------
----o--o---o-o-----------o---o------------o-------o----o--o--o--o-o---------------o-----------------
-------oo--o-o-----o-----o----o-o--o----------------------o-------o------o----oo----ooo---------o---
o-----oo-------------------o--o-----o-----------o------o-------o----o-----------o----------------o--
--o---o-------o------------o--------------------o----o--o-------------oo---o---------oo--------o----
--o--------o---------o------------o------o-------o------------o-------o---o---------ooooo-----------
------o--------------o-o-o---------o---o-------o--o-----o-------o-o----------o-----oo-ooo----------o
--o---------------o----o--oo-------------o---------o-------------------oo---------oo-o-ooo----------
-o-----------o------ooo----o----------------ooo-----o--------o--o---o-----------o-o-oooooo--------oo
-o---o-------o---o-oooo-----o-------------------o----oo-----------------o--o--------o--o------o--o--
-------o---o------oooooo--o----ooo--o--------o-------o----------------------------oo-oo-o--o--------
o--oo------o-----oo--o-oo------------oo--o------o--o-------------oo----o------------oooo-o------oo--
-----o----------ooooooooo--------------oo--------------oo-----o-----o-o--o------o----------o----o---"
        />
      </div>
      <button
        onClick={[Function]}
      >
        process Input
      </button>
      <button
        onClick={[Function]}
      >
        Clean noise
      </button>
    </section>
    <div
      className="results-panel"
    >
      <Results
        state={
          Object {
            "definition": 0.8,
            "input": "
----o--oo----o--ooo--ooo---------o---oo-o----oo---o--o---------o----o------o----------------o--o--o-
--o-o-----oooooooo-oooooo-------o----o------ooo-o---o--o----o------o--o---ooo-----o--oo-o------o----
--o--------oo-ooo-oo-oo-oo------------------ooooo-----o-----o------o---o--o--o-o-o------o----o-o-o--
-------o--oooooo--o-oo-o--o-o-----oo--o-o-oo--o-oo-oo-o--------o-----o------o-ooooo---o--o--o-------
------o---o-ooo-ooo----o------o-------o---oo-ooooo-o------o----o--------o-oo--ooo-oo-------------o-o
-o--o-----o-o---o-ooooo-o-------o----o---------o-----o-oo-----------oo----ooooooo-ooo-oo------------
o-------------ooooo-o--o--o--o-------o--o-oo-oo-o-o-o----o-------------o--oooo--ooo-o----o-----o--o-
--o-------------------------oo---------oo-o-o--ooo----o-----o--o--o----o--o-o-----o-o------o-o------
-------------------o-----------------o--o---------------o--------o--oo-o-----oo-oo---o--o---o-----oo
----------o----------o------------------o--o----o--o-o------------oo------o--o-o---o-----o----------
------o----o-o---o-----o-o---------oo-o--------o---------------------------------o-o-o--o-----------
---------------o-------------o-------o-------------------o-----o---------o-o-------------o-------oo-
-o--o-------------o-o--------o--o--oo-------------o----ooo----o-------------o----------oo----o---o-o
-o--o-------------o----oo------o--o-------o--o-----------o----o-----o--o----o--oo-----------o-------
-o-----oo-------o------o---------------o--o----------o-----o-------o-----------o---o-o--oooooo-----o
-o--------o-----o-----o---------oo----oo---o-----------o---o--oooo-oo--o-------o------oo--oo--o-----
------------o-------------------o----oooo-------------oo-oo-----ooo-oo-----o-------o-oo-oooooooo---o
-----------------------------------oooooooo---o-----o-------o--oooooo-o------------o-o-ooooooo-o----
------------o------o-------o-------oo-oo--o--o---------o--o-o-o-ooooo-o--------------oo-o----o-oo-o-
---o-o----------o--------oo----o----oooooooo-------o----o-o-o-o-----o-o-----o----------ooo-oo--o---o
-o-o---------o-o---------------o--o--o--ooo---ooo-------o------oo-oo------------o--------o--o-o--o--
-------oo---------------------------o-oo----------o------o-o-------o-----o----o-----o-oo-o-----o---o
---o--------o-----o-------o-oo-----oo--oo-o----oo----------o--o---oo------oo----o-----o-------o-----
---o--ooo-o---------o-o----o------------o---------o----o--o-------o-------------o----------------oo-
---o------o----------------o----o------o------o---oo-----------o-------------o----------oo---------o
--oo---------------o--o------o---o-----o--o-------------o------o-------o-----o-----o----o------o--o-
-o-------o----------o-o-o-------o-----o--o-o-----------o-oo-----------o------o---------o-----o-o----
----------o----o-------o----o--o------o------------o---o---------------oo----o-----ooo--------------
----o--------oo----o-o----o--o------ooo----o-oooo---o--o-oo--------o-oo-----o-o---o-o--o-----oo-----
------o--------o-ooooo----o---o--o-----o---------------o-o-------o-----o----------------------------
o-------oo----o--oooooo-o---o--o------oooo----------o-oo-------o---o----------o------oo-------------
-o---o----------o--oo-oo-o---o-----o-o-----------------------oo--o------o------o--------------------
-----oo-o-o-o---ooooooooo----o----o--------o--o---oo---o------------o----------o-o---o------o-o--oo-
------o------o---ooo-o---------------------------o--o---o---o----o--o-------o-----o------o----o----o
-------o----------ooo-o-----o----o---o--o-oo--o--o-o--o------o--o-oo---ooo------------------------o-
-o-------o------o-o--ooo--o---o---oo-----o----o-------------o----o-ooo-o------o--o-o------o-o-------
---oo--o---o-o---------o---o--------------o--o-----o-------o-----o--o---o-oo--------o----o----o-----
o------o----oo-o-----------oo--o---o--------o-o------o-------o-o------o-oo---------o-----oo---------
----o--o---o-o-----------o---o------------o-------o----o--o--o--o-o---------------o-----------------
-------oo--o-o-----o-----o----o-o--o----------------------o-------o------o----oo----ooo---------o---
o-----oo-------------------o--o-----o-----------o------o-------o----o-----------o----------------o--
--o---o-------o------------o--------------------o----o--o-------------oo---o---------oo--------o----
--o--------o---------o------------o------o-------o------------o-------o---o---------ooooo-----------
------o--------------o-o-o---------o---o-------o--o-----o-------o-o----------o-----oo-ooo----------o
--o---------------o----o--oo-------------o---------o-------------------oo---------oo-o-ooo----------
-o-----------o------ooo----o----------------ooo-----o--------o--o---o-----------o-o-oooooo--------oo
-o---o-------o---o-oooo-----o-------------------o----oo-----------------o--o--------o--o------o--o--
-------o---o------oooooo--o----ooo--o--------o-------o----------------------------oo-oo-o--o--------
o--oo------o-----oo--o-oo------------oo--o------o--o-------------oo----o------------oooo-o------oo--
-----o----------ooooooooo--------------oo--------------oo-----o-----o-o--o------o----------o----o---",
            "invaders": Array [
              Spider {
                "height": 8,
                "image": "--o-----o--
---o---o---
--ooooooo--
-oo-ooo-oo-
ooooooooooo
o-ooooooo-o
o-o-----o-o
---oo-oo---",
                "map": Array [
                  Array [
                    "-",
                    "-",
                    "o",
                    "-",
                    "-",
                    "-",
                    "-",
                    "-",
                    "o",
                    "-",
                    "-",
                  ],
                  Array [
                    "-",
                    "-",
                    "-",
                    "o",
                    "-",
                    "-",
                    "-",
                    "o",
                    "-",
                    "-",
                    "-",
                  ],
                  Array [
                    "-",
                    "-",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "-",
                    "-",
                  ],
                  Array [
                    "-",
                    "o",
                    "o",
                    "-",
                    "o",
                    "o",
                    "o",
                    "-",
                    "o",
                    "o",
                    "-",
                  ],
                  Array [
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                  ],
                  Array [
                    "o",
                    "-",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "-",
                    "o",
                  ],
                  Array [
                    "o",
                    "-",
                    "o",
                    "-",
                    "-",
                    "-",
                    "-",
                    "-",
                    "o",
                    "-",
                    "o",
                  ],
                  Array [
                    "-",
                    "-",
                    "-",
                    "o",
                    "o",
                    "-",
                    "o",
                    "o",
                    "-",
                    "-",
                    "-",
                  ],
                ],
                "name": "spider",
                "negativesList": Array [
                  Array [
                    0,
                    0,
                  ],
                  Array [
                    0,
                    1,
                  ],
                  Array [
                    0,
                    3,
                  ],
                  Array [
                    0,
                    4,
                  ],
                  Array [
                    0,
                    5,
                  ],
                  Array [
                    0,
                    6,
                  ],
                  Array [
                    0,
                    7,
                  ],
                  Array [
                    0,
                    9,
                  ],
                  Array [
                    0,
                    10,
                  ],
                  Array [
                    1,
                    0,
                  ],
                  Array [
                    1,
                    1,
                  ],
                  Array [
                    1,
                    2,
                  ],
                  Array [
                    1,
                    4,
                  ],
                  Array [
                    1,
                    5,
                  ],
                  Array [
                    1,
                    6,
                  ],
                  Array [
                    1,
                    8,
                  ],
                  Array [
                    1,
                    9,
                  ],
                  Array [
                    1,
                    10,
                  ],
                  Array [
                    2,
                    0,
                  ],
                  Array [
                    2,
                    1,
                  ],
                  Array [
                    2,
                    9,
                  ],
                  Array [
                    2,
                    10,
                  ],
                  Array [
                    3,
                    0,
                  ],
                  Array [
                    3,
                    3,
                  ],
                  Array [
                    3,
                    7,
                  ],
                  Array [
                    3,
                    10,
                  ],
                  Array [
                    5,
                    1,
                  ],
                  Array [
                    5,
                    9,
                  ],
                  Array [
                    6,
                    1,
                  ],
                  Array [
                    6,
                    3,
                  ],
                  Array [
                    6,
                    4,
                  ],
                  Array [
                    6,
                    5,
                  ],
                  Array [
                    6,
                    6,
                  ],
                  Array [
                    6,
                    7,
                  ],
                  Array [
                    6,
                    9,
                  ],
                  Array [
                    7,
                    0,
                  ],
                  Array [
                    7,
                    1,
                  ],
                  Array [
                    7,
                    2,
                  ],
                  Array [
                    7,
                    5,
                  ],
                  Array [
                    7,
                    8,
                  ],
                  Array [
                    7,
                    9,
                  ],
                  Array [
                    7,
                    10,
                  ],
                ],
                "positivesList": Array [
                  Array [
                    0,
                    2,
                  ],
                  Array [
                    0,
                    8,
                  ],
                  Array [
                    1,
                    3,
                  ],
                  Array [
                    1,
                    7,
                  ],
                  Array [
                    2,
                    2,
                  ],
                  Array [
                    2,
                    3,
                  ],
                  Array [
                    2,
                    4,
                  ],
                  Array [
                    2,
                    5,
                  ],
                  Array [
                    2,
                    6,
                  ],
                  Array [
                    2,
                    7,
                  ],
                  Array [
                    2,
                    8,
                  ],
                  Array [
                    3,
                    1,
                  ],
                  Array [
                    3,
                    2,
                  ],
                  Array [
                    3,
                    4,
                  ],
                  Array [
                    3,
                    5,
                  ],
                  Array [
                    3,
                    6,
                  ],
                  Array [
                    3,
                    8,
                  ],
                  Array [
                    3,
                    9,
                  ],
                  Array [
                    4,
                    0,
                  ],
                  Array [
                    4,
                    1,
                  ],
                  Array [
                    4,
                    2,
                  ],
                  Array [
                    4,
                    3,
                  ],
                  Array [
                    4,
                    4,
                  ],
                  Array [
                    4,
                    5,
                  ],
                  Array [
                    4,
                    6,
                  ],
                  Array [
                    4,
                    7,
                  ],
                  Array [
                    4,
                    8,
                  ],
                  Array [
                    4,
                    9,
                  ],
                  Array [
                    4,
                    10,
                  ],
                  Array [
                    5,
                    0,
                  ],
                  Array [
                    5,
                    2,
                  ],
                  Array [
                    5,
                    3,
                  ],
                  Array [
                    5,
                    4,
                  ],
                  Array [
                    5,
                    5,
                  ],
                  Array [
                    5,
                    6,
                  ],
                  Array [
                    5,
                    7,
                  ],
                  Array [
                    5,
                    8,
                  ],
                  Array [
                    5,
                    10,
                  ],
                  Array [
                    6,
                    0,
                  ],
                  Array [
                    6,
                    2,
                  ],
                  Array [
                    6,
                    8,
                  ],
                  Array [
                    6,
                    10,
                  ],
                  Array [
                    7,
                    3,
                  ],
                  Array [
                    7,
                    4,
                  ],
                  Array [
                    7,
                    6,
                  ],
                  Array [
                    7,
                    7,
                  ],
                ],
                "width": 11,
              },
              Squid {
                "height": 8,
                "image": "---oo---
--oooo--
-oooooo-
oo-oo-oo
oooooooo
--o--o--
-o-oo-o-
o-o--o-o",
                "map": Array [
                  Array [
                    "-",
                    "-",
                    "-",
                    "o",
                    "o",
                    "-",
                    "-",
                    "-",
                  ],
                  Array [
                    "-",
                    "-",
                    "o",
                    "o",
                    "o",
                    "o",
                    "-",
                    "-",
                  ],
                  Array [
                    "-",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "-",
                  ],
                  Array [
                    "o",
                    "o",
                    "-",
                    "o",
                    "o",
                    "-",
                    "o",
                    "o",
                  ],
                  Array [
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                    "o",
                  ],
                  Array [
                    "-",
                    "-",
                    "o",
                    "-",
                    "-",
                    "o",
                    "-",
                    "-",
                  ],
                  Array [
                    "-",
                    "o",
                    "-",
                    "o",
                    "o",
                    "-",
                    "o",
                    "-",
                  ],
                  Array [
                    "o",
                    "-",
                    "o",
                    "-",
                    "-",
                    "o",
                    "-",
                    "o",
                  ],
                ],
                "name": "squid",
                "negativesList": Array [
                  Array [
                    0,
                    0,
                  ],
                  Array [
                    0,
                    1,
                  ],
                  Array [
                    0,
                    2,
                  ],
                  Array [
                    0,
                    5,
                  ],
                  Array [
                    0,
                    6,
                  ],
                  Array [
                    0,
                    7,
                  ],
                  Array [
                    1,
                    0,
                  ],
                  Array [
                    1,
                    1,
                  ],
                  Array [
                    1,
                    6,
                  ],
                  Array [
                    1,
                    7,
                  ],
                  Array [
                    2,
                    0,
                  ],
                  Array [
                    2,
                    7,
                  ],
                  Array [
                    3,
                    2,
                  ],
                  Array [
                    3,
                    5,
                  ],
                  Array [
                    5,
                    0,
                  ],
                  Array [
                    5,
                    1,
                  ],
                  Array [
                    5,
                    3,
                  ],
                  Array [
                    5,
                    4,
                  ],
                  Array [
                    5,
                    6,
                  ],
                  Array [
                    5,
                    7,
                  ],
                  Array [
                    6,
                    0,
                  ],
                  Array [
                    6,
                    2,
                  ],
                  Array [
                    6,
                    5,
                  ],
                  Array [
                    6,
                    7,
                  ],
                  Array [
                    7,
                    1,
                  ],
                  Array [
                    7,
                    3,
                  ],
                  Array [
                    7,
                    4,
                  ],
                  Array [
                    7,
                    6,
                  ],
                ],
                "positivesList": Array [
                  Array [
                    0,
                    3,
                  ],
                  Array [
                    0,
                    4,
                  ],
                  Array [
                    1,
                    2,
                  ],
                  Array [
                    1,
                    3,
                  ],
                  Array [
                    1,
                    4,
                  ],
                  Array [
                    1,
                    5,
                  ],
                  Array [
                    2,
                    1,
                  ],
                  Array [
                    2,
                    2,
                  ],
                  Array [
                    2,
                    3,
                  ],
                  Array [
                    2,
                    4,
                  ],
                  Array [
                    2,
                    5,
                  ],
                  Array [
                    2,
                    6,
                  ],
                  Array [
                    3,
                    0,
                  ],
                  Array [
                    3,
                    1,
                  ],
                  Array [
                    3,
                    3,
                  ],
                  Array [
                    3,
                    4,
                  ],
                  Array [
                    3,
                    6,
                  ],
                  Array [
                    3,
                    7,
                  ],
                  Array [
                    4,
                    0,
                  ],
                  Array [
                    4,
                    1,
                  ],
                  Array [
                    4,
                    2,
                  ],
                  Array [
                    4,
                    3,
                  ],
                  Array [
                    4,
                    4,
                  ],
                  Array [
                    4,
                    5,
                  ],
                  Array [
                    4,
                    6,
                  ],
                  Array [
                    4,
                    7,
                  ],
                  Array [
                    5,
                    2,
                  ],
                  Array [
                    5,
                    5,
                  ],
                  Array [
                    6,
                    1,
                  ],
                  Array [
                    6,
                    3,
                  ],
                  Array [
                    6,
                    4,
                  ],
                  Array [
                    6,
                    6,
                  ],
                  Array [
                    7,
                    0,
                  ],
                  Array [
                    7,
                    2,
                  ],
                  Array [
                    7,
                    5,
                  ],
                  Array [
                    7,
                    7,
                  ],
                ],
                "width": 8,
              },
            ],
            "results": Array [],
            "samples": Array [],
          }
        }
      />
    </div>
  </ContextProvider>
</div>
`;
